src/store/Configuration/actions.ts

```diff
@@ -0,0 +1,32 @@
+import axios from '@/axios-auth'
+
+export default {
+  /** Fetches stateFiling from the API and, if successful, triggers mutation. */
+  fetchConfiguration (context, applicationUrl: string): Promise<any> {
+    // fetch config from API
+    // eg, http://localhost:8080/business/config/configuration.json
+    // eg, https://dev.bcregistry.ca/business/config/configuration.json
+    const url = `${applicationUrl}config/configuration.json`
+    const headers = {
+      'Accept': 'application/json',
+      'ResponseType': 'application/json',
+      'Cache-Control': 'no-store'
+    }
+    return new Promise((resolve, reject) => {
+      axios.get(url, { headers })
+        .then((response) => {
+          if (!response.data) {
+            reject(new Error('Invalid configuration.json'))
+          } else {
+            context.commit('setConfiguration', response.data)
+            context.commit('setSessionVariables', response.data) // For backward compatibility
+            context.commit('setAxiosBaseUrl', context.getters.getLegalApiUrl)
+            resolve(response.data)
+          }
+        })
+        .catch((error) => {
+          reject(new Error('Error: ' + error))
```

Comment:
Redundant?

Reply:
Good question, I had to think about this!  My understanding is that the "error" on line 27 (above) is from the Axios promise, so I want to expose that error.  I'm happy to be corrected if I've got this wrong and it's not needed.

Reply:
If you don't catch that error here, it will still be exposed by this method's caller (since you're essentially re-throwing the error).

This code, I believe, will display the error twice: once because it's caught here, and another time when it's caught (or not) in the caller. Trigger this error then check the developer console.

Reply:
I'm also 99% sure you don't have to create a new promise or resolve or reject it in this method: if you simply return a value (in the axios.get then block) then it gets wrapped in a promise resolve (since the method is async), and if you throw an error then it gets wrapped in a promise reject.

Try it. It's a nice simplification. Promise === Async.

Reply:
Sure, I'm happy to give it a try.  Is this what you're thinking?
```
return axios.get(url, { headers })
      .then((response) => {
        if (!response.data) {
          return new Error('Invalid configuration.json')
        } else {
          context.commit('setConfiguration', response.data)
          context.commit('setSessionVariables', response.data)
          context.commit('setAxiosBaseUrl', context.getters.getLegalApiUrl)
          return response.data
        }
      })
```

Reply:
Yes! Does it work as you expect?

Reply:
If returning a Promise is needed because this method is called via the store, please add a comment. (We talked about this. I'm just leaving this comment here to remind myself to go back and check your comment.)

Reply:
Thanks for the comment.